resources/src/trade.rs

```diff
@@ -0,0 +1,253 @@
+use chrono::prelude::*;
+use amount::Amount;
+// use asset::AssetIdentifier;
+use offer::PriceRatio;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+/// A trade represents an offer that was fulfilled between two assets and accounts.
+///
+/// https://www.stellar.org/developers/horizon/reference/resources/trade.html
+#[derive(Debug)]
+pub struct Trade {
+    id: String,
+    offer_id: String,
+    paging_token: String,
+    ledger_close_time: DateTime<Utc>,
+    base_account: String,
+    base_amount: Amount,
+    // base_asset: AssetIdentifier,
+    counter_amount: Amount,
+    counter_account: String,
+    // counter_asset: AssetIdentifier,
+    price: PriceRatio,
+    seller: Seller,
+}
+
+/// The seller involved in the trade.
+#[derive(Eq, PartialEq, Debug, Copy, Clone)]
+pub enum Seller {
+    /// The base created the sell offer.
+    Base,
+    /// The counter created the sell offer.
+    Counter,
+}
+
+impl Seller {
+    /// Returns whether the seller is the base
+    ///
+    /// ```
+    /// use stellar_resources::TradeSeller;
+    /// let seller = TradeSeller::Base;
+    /// assert!(seller.is_base());
+    /// ```
+    pub fn is_base(&self) -> bool {
+        *self == Seller::Base
+    }
+
+    /// Returns whether the seller is the counter
+    ///
+    /// ```
+    /// use stellar_resources::TradeSeller;
+    /// let seller = TradeSeller::Counter;
+    /// assert!(seller.is_counter());
+    /// ```
+    pub fn is_counter(&self) -> bool {
+        *self == Seller::Counter
+    }
+}
+
+impl<'de> Deserialize<'de> for Trade {
+    fn deserialize<D>(d: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let rep: TradeRepresentation = TradeRepresentation::deserialize(d)?;
```

Comment:
WOW

Reply:
I used the keyword "intermediate" and you are using the keyword "representation" I think both are fine, "Denormalized" might be more accurate than both, but we should pick one and stick with it.  We can also address this in a future pr.